sdk/tables/azure-data-tables/azure/data/tables/aio/_base_client_async.py

```diff
@@ -28,44 +17,41 @@
     HttpLoggingPolicy,
     UserAgentPolicy,
     ProxyPolicy,
-    AzureSasCredentialPolicy
+    AzureSasCredentialPolicy,
+    RequestIdPolicy,
+    CustomHookPolicy,
+    NetworkTraceLoggingPolicy
 )
 from azure.core.pipeline.transport import (
     AsyncHttpTransport,
     HttpRequest,
 )
 
+from .._generated.aio import AzureTable
+from .._base_client import AccountHostsMixin
 from .._authentication import SharedKeyCredentialPolicy
-from .._constants import STORAGE_OAUTH_SCOPE, CONNECTION_TIMEOUT, READ_TIMEOUT
-from .._generated.aio._configuration import AzureTableConfiguration
+from .._constants import STORAGE_OAUTH_SCOPE
 from .._models import BatchErrorException, BatchTransactionResult
-from .._policies import (
-    CosmosPatchTransformPolicy,
-    StorageContentValidation,
-    StorageRequestHook,
-    StorageHosts,
-    StorageHeadersPolicy,
-    StorageLoggingPolicy,
-)
+from .._policies import StorageHosts, StorageHeadersPolicy
 from .._sdk_moniker import SDK_MONIKER
-from ._policies_async import (
-    AsyncStorageResponseHook,
-    AsyncTablesRetryPolicy
-)
-
-if TYPE_CHECKING:
-    from azure.core.pipeline import Pipeline
-    from azure.core.configuration import Configuration
-
-_LOGGER = logging.getLogger(__name__)
+from ._policies_async import AsyncTablesRetryPolicy
 
 
-class AsyncStorageAccountHostsMixin(object):
-    def __enter__(self):
-        raise TypeError("Async client only supports 'async with'.")
+class AsyncTablesBaseClient(AccountHostsMixin):
 
-    def __exit__(self, *args):
-        pass
+    def __init__(
+        self,
+        account_url,  # type: str
+        credential=None,  # type: str
+        **kwargs  # type: Any
+    ):
+        # type: (...) -> None
+        super(AsyncTablesBaseClient, self).__init__(account_url, credential=credential, **kwargs)
+        self._client = AzureTable(
+            self.url,
+            policies=kwargs.pop('policies', self._policies),
+            **kwargs
+        )
 
     async def __aenter__(self):
         await self._client.__aenter__()
```

Comment:
Because this inherits from the sync base client, will there be a weird behavior where you can use an async client as a sync context manager:
```python
with TableClient(...) as tc:
    await tc.get_entity(...)
```

Reply:
Nope - the async client does _not_ inherit from the sync base client. For exactly this reason. It only inherits from the AccountHostsMixin